src/V3Sched.cpp

```diff
@@ -0,0 +1,964 @@
+            m_updates.push_back(new AstAssign{flp, refp, new AstConst{flp, AstConst::BitFalse{}}});
+            return {currp(), false};
+        }
+        default:  // LCOV_EXCL_START
+            senItemp->v3fatalSrc("Unknown edge type");
+            return {nullptr, false};
+        }  // LCOV_EXCL_STOP
+    }
+
+public:
+    // Returns the expression computing the trigger, and a bool indicating that
+    // this trigger should be fired on the first evaluation (at initialization)
+    std::pair<AstNode*, bool> build(const AstSenTree* senTreep) {
+        FileLine* const flp = senTreep->fileline();
+        AstNode* resultp = nullptr;
+        bool firedAtInitialization = false;
+        for (AstSenItem* senItemp = senTreep->sensesp(); senItemp;
+             senItemp = VN_AS(senItemp->nextp(), SenItem)) {
+            const auto& pair = createTerm(senItemp);
+            if (AstNode* const termp = pair.first) {
+                resultp = resultp ? new AstOr{flp, resultp, termp} : termp;
+                firedAtInitialization |= pair.second;
+            }
+        }
+        return {resultp, firedAtInitialization};
+    }
+
+    std::vector<AstNodeStmt*> getAndClearUpdates() {
+        m_hasUpdate.clear();
+        return std::move(m_updates);
+    }
+
+    // CONSTRUCTOR
+    SenExprBuilder(AstNetlist* netlistp, AstCFunc* initp)
+        : m_initp{initp}
+        , m_scopeTopp{netlistp->topScopep()->scopep()}
+        , m_dpiExportTriggerp{netlistp->dpiExportTriggerp()} {}
+};
+
+std::vector<const AstSenTree*> getSenTreesUsedBy(std::vector<const LogicByScope*> lbsps) {
+    const VNUser1InUse user1InUse;
+    std::vector<const AstSenTree*> result;
+    for (const LogicByScope* const lbsp : lbsps) {
+        lbsp->foreach ([&](AstScope*, AstActive* activep) {
+            AstSenTree* const senTreep = activep->sensesp();
+            if (senTreep->user1SetOnce()) return;
+            if (senTreep->hasClocked() || senTreep->hasHybrid()) result.push_back(senTreep);
+        });
+    }
+    return result;
+}
+
+AstAssign* setVar(AstVarScope* vscp, uint32_t val) {
+    FileLine* const flp = vscp->fileline();
+    AstVarRef* const refp = new AstVarRef{flp, vscp, VAccess::WRITE};
+    AstConst* const zerop = new AstConst{flp, AstConst::DtypedValue{}, vscp->dtypep(), val};
+    return new AstAssign{flp, refp, zerop};
+};
+
+struct TriggerKit {
+    // The TRIGGERVEC AstVarScope representing these trigger flags
+    AstVarScope* const m_vscp;
+    // The AstCFunc that computes the current active triggers
+    AstCFunc* const m_funcp;
+    // The AstCFunc that dumps the current active triggers
+    AstCFunc* const m_dumpp;
+    // The map from input sensitivity list to trigger sensitivity list
+    const std::unordered_map<const AstSenTree*, AstSenTree*> m_map;
+
+    VL_UNCOPYABLE(TriggerKit);
+
+    // Create an AstSenTree that is sensitive to trigger 0. Must not exist yet!
```

Comment:
Suggest a constexpr for '0' to indicate this trigger value usage, used at appropriate points.